src/parser.rs

```diff
@@ -283,11 +283,30 @@ impl<'a> Parser<'a> {
                 },
             }, // End of Token::Word
             Token::Mult => Ok(Expr::Wildcard),
-            tok @ Token::Minus | tok @ Token::Plus => {
-                let op = if tok == Token::Plus {
-                    UnaryOperator::Plus
-                } else {
-                    UnaryOperator::Minus
+            tok @ Token::Plus
+            | tok @ Token::Minus
+            | tok @ Token::DoubleExclamationMark
+            | tok @ Token::PGSquareRoot
+            | tok @ Token::PGCubeRoot
+            | tok @ Token::AtSign
+            | tok @ Token::Tilde => {
+                let op = match tok {
+                    Token::Plus => UnaryOperator::Plus,
+                    Token::Minus => UnaryOperator::Minus,
+                    Token::DoubleExclamationMark if dialect_of!(self is PostgreSqlDialect) => {
+                        UnaryOperator::PGPrefixFactorial
+                    }
```

Comment:
If we keep the dialect-specific parsing, it probably would be clearer to handle +/- in a separate case from the PG-specific stuff. That way you would be able to specify the `if dialect_of` only once instead of repeating it for every operator.

Reply:
Were you planning to address this and maybe adding GenericDialect to the list of dialects that support the new operations? Or can I merge this as is?

Reply:
addressed